### DISCUSSION TITLE
Question about odmfuse.sh

### DISCUSSION THREAD

**Discussion by danielwangksu** [OE4T/meta-tegra#627] [Q&A] [answered]
Hi all,

I'm still using the warrior branch and recently encountered an issue when I tried to burn efuse through the odmfuse.sh script. It seems odmfuse.sh requires a specific directory structure, such as bootloader, kernel, pkc, etc. folders have to existing. And it requires configure file such as: jetson-xavier.conf, jetson-tx2i.conf (which indirectly depends on other files in specific locations). My question is how can I use the script to burn efuse with the Yocto output ZIP file?

Is there a update in the latest change helps this requirement? My current workflow is to get the ZIP file from Yocto and using my standalone scripts (based on tegra194_doflash.sh and tegra194_flash_helper.sh) for signing and encryption because I do not want to leave the private key and encryption key in Yocto code base. It has been working fine. But I'm having trouble to find a way to burn the efuse.

Thank you for your help and any advice would be appreciated!

**Comment by madisongh** [comment, marked as answer]
You could use the `--noburn` option to generate the "manufacturing" fuse burning package (out of a stock L4T setup), and just keep that around for doing the fusing.

With recent branches here, though, you can extract the `odmfuse_pkc.xml` file from the manufacturing fusing package and set the ODMFUSE_FILE variable to point to it in a bbappend on the `tegra-bootfiles` recipe.  If that file is present, the tegraflash package that gets generated will include a `burnfuses.sh` script for performing the fuse burning.

**Reply by danielwangksu** [reply]
Thank you @madisongh! Glad to know that I can use the L4T setup for generating efuse burning package. I will give the second method a try.
